base/traits.jl

```diff
@@ -20,3 +20,40 @@ TypeArithmetic(instance) = TypeArithmetic(typeof(instance))
 TypeArithmetic(::Type{<:AbstractFloat}) = ArithmeticRounds()
 TypeArithmetic(::Type{<:Integer}) = ArithmeticOverflows()
 TypeArithmetic(::Type{<:Any}) = ArithmeticUnknown()
+
+# trait for objects that support ranges with regular step
+"""
+    TypeRangeStep(instance)
+    TypeRangeStep(T::Type)
```

Comment:
Good question. The other traits are not exported, but I'm not sure what their use outside of Base is. @timholy?

The traits could also be mentioned somewhere in the manual, but I'm not sure where. They could go to the Interfaces section, but they aren't related to any existing interface (except maybe `Real`). Maybe we could have a sub-section listing all traits that a custom type may want to implement: this trait will generally be useful when you write your own type.